Formula/python@2.rb

```diff
@@ -104,14 +104,16 @@ def install
     ldflags  = []
     cppflags = []
 
-    unless MacOS::CLT.installed?
-      # Help Python's build system (setuptools/pip) to build things on Xcode-only systems
+    if !MacOS::CLT.installed? || !MacOS::CLT.headers_installed?
+      # Help Python's build system (setuptools/pip) to build things on SDK-based systems
       # The setup.py looks at "-isysroot" to get the sysroot (and not at --sysroot)
-      cflags   << "-isysroot #{MacOS.sdk_path}"
-      ldflags  << "-isysroot #{MacOS.sdk_path}"
-      cppflags << "-I#{MacOS.sdk_path}/usr/include" # find zlib
+      cflags  << "-isysroot #{MacOS.sdk_path}"
+      ldflags << "-isysroot #{MacOS.sdk_path}"
+      cflags  << "-I/usr/include" # find zlib
```

Comment:
I think this is currently broken on Xcode-only installations. There are two problems here:

1. Python 2 requires this in the `CFLAGS` specifically (Python 3 appears to be happy with `CPPFLAGS` as written).
2. The check that Python's machinery does is relative to the `SYSROOT`, so interpolating in `MacOS.sdk_path` breaks it.